src/main/java/uk/gov/hmcts/reform/sendletter/tasks/MarkLettersPostedTask.java

```diff
@@ -93,20 +92,20 @@ public void run() {
     }
 
     private void markAsPosted(LetterPrintStatus letterPrintStatus, String reportFileName) {
-        Optional<Letter> optional = repo.findById(letterPrintStatus.id);
+        Optional<LetterStatus> optional = repo.findLetterStatus(letterPrintStatus.id);
```

Comment:
There's no need to query the DB for the status first. Instead, we can just update the record (`... where id = ... and status == 'Uploaded'`. If the query returns 1 (which is approx 100% cases), done. If not, then we can read the record and log the necessary details.

Reply:
This change assimilates the current behaviour. Would be way better to do bulk update on such task like printing flag update. Would even consider skipping the extra branch reporting on bad status and keep it for stale letter report instead

Reply:
Fair enough

Reply:
Shame, information where `printed_at` exists is retrieved from report file. So.. per letter :(